tests/kernel/kselftests.pm

```diff
@@ -14,24 +14,39 @@ use testapi;
 use warnings;
 use serial_terminal 'select_serial_terminal';
 use LTP::WhiteList;
+use version_utils qw(is_transactional);
+use transactional 'trup_install';
+
+sub download_kernel_source
+{
+    my @kv = split /\./, script_output "uname -r";
+    my ($kv0, $kv1, $kv2) = ($kv[0], $kv[1], (split /-/, $kv[2])[0]);
+    my $url = "https://mirrors.edge.kernel.org/pub/linux/kernel/v$kv0.x/linux-$kv0.$kv1";
+
+    $url .= $kv2 eq '0' ? ".tar.gz" : ".$kv2.tar.gz";
+
+    assert_script_run("mkdir /root/linux");
+    assert_script_run("curl " . $url . "| tar xz --strip-components=1 -C /root/linux", timeout => 600);
```

Comment:
There's no kselftests there

Reply:
> There's no kselftests there

@acerv @czerw That's not true at least for Tumbleweed (I haven't checked Alp yet):
```
$ rpm -q kernel-source
kernel-source-6.5.6-1.1.noarch

$ rpm -ql kernel-source|grep tools/testing/selftests
/usr/src/linux-6.5.6-1/tools/testing/selftests/zram/zram_lib.sh
...
```
=> IMHO it would be faster to download kernel sources than clone them from git.

Reply:
Thanks, if needed I can add this support

Reply:
Reasons to add it:
1) Installation from package will be probably faster than clone git.
2) More recent code by default (i.e. more fixes from mainline): For Tumbleweed and Alp we download https://mirrors.kernel.org/pub/linux/kernel/v6.x/linux-6.5.tar.gz, but in sources we have currently 6.5.6 (and it will keep update). Also for SLES we will have more recent code 5.14 vs. 5.14.21, but this will stay the same.
3) We can add patches to backport fixes (if we bother to fix kselftests).